src/llamacrawl/services/exceptions.py

```
@@ -0,0 +1,92 @@
+    """Raised when source is already being ingested.
+
+    This exception indicates that a distributed lock is already held for the
+    specified source, preventing concurrent ingestion operations.
+
+    Example:
+        When attempting to ingest GitHub data while another process is already
+        running GitHub ingestion, this error will be raised to prevent duplicate
+        processing and potential data corruption.
+    """
+
+    pass
```

Ruff (0.14.0)

Unnecessary `pass` statement

Remove unnecessary `pass`

(PIE790)